examples/pipelines/guestbook-resources.yaml

```diff
@@ -1,30 +1,43 @@
 apiVersion: pipeline.knative.dev/v1alpha1
 kind: Resource
 metadata:
-  name: guestbook-resources-sample
+  name: guestbook-resources-git
   namespace: default
 spec:
-  resources:
-    - name: guestbook
-      type: git
-      params:
+    name: guestbook
+    type: git
+    params:
+    - name: url
+      value: github.com/kubernetes/examples
+    - name: revision
+      value: HEAD      
+    - name: serviceAccount
+      value: githubServiceAccount
+---      
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: Resource
```

Comment:
interesting! so one would need to reference each `Resource` separately?

I am fine with doing this but I assumed differently in the examples I wrote out, e.g. in https://github.com/knative/build-pipeline/issues/64 I assumed you could declare one `Resource` type which contained a list of all the resources you needed for your Pipeline. One thing I like about that is that it means the number of yaml files a person would need to interact with their Pipeline is limited.

But if it makes more sense to have a separate object for each `Resource` we can update the issues!

Reply:
I made this change as part of this PR actually, its because when I started working on the git resource I found that having an array in the Spec that refers to another object makes the code too awkward